npc/merchants/advanced_refiner.txt

```diff
@@ -44,14 +44,22 @@
 //=========================================================================
 
 payon,157,146,6	script	Suhnbi#cash	4_M_03,{
-	disable_items;
-	mes "[Suhnbi]";
-	mes "I am the Armsmith";
-	mes "I can refine all kinds of weapons,";
-	mes "armor and equipment, so let me";
-	mes "know what you want to refine.";
-	next;
+	mes("[Suhnbi]");
+	mes("I am the Armsmith");
+	mes("I can refine all kinds of weapons,");
+	mes("armor and equipment, so let me");
+	mes("know what you want to refine.");
 
+	if (getbattleflag("features/replace_refine_npcs") == 1) {
+		if (openrefineryui())
+			close();
```

Comment:
In official, when you click the npc, The ui will show after you click the `close` button.

Reply:
I see no reason to replicate the behavior, The command is designed to return false in-case of failure, and to fallback to actual NPCs and I'd rather keep the decent design over blind replica.

Reply:
> > When the UI is active, you cannot open boxes, use items, use skills, put items on cart.
> > PS: When using skill it will show this message.
> > ![image](https://user-images.githubusercontent.com/4826387/56849568-e1d5bc00-6928-11e9-8fc7-1dd37619c806.png)
> 
> This is already happening with the current code, no?
> I will look into this message.

No.